### PR TITLE
 Bump fmtlib from 7.0.0 to 8.1.1

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Bump fmtlib from 7.0.0 to 8.1.1 (with compile time format support).

One of the most major change is that now fmt will check formatted string at runtime, it will be used for logging (but not for Exception ctor, since there is no ability to do so) and will catch possible format issues at compile time, and existing code already has few issues that had been fixed in this PR (see commit history for more details).

### CODE REVIEW

**Comment by azat** [comment]
>[ClickHouse special build check (actions) — 6/7 builds are OK](https://s3.amazonaws.com/clickhouse-builds/34223/d0d71180dd7b89d5c3eab11c59a8355803cbe7bc/clickhouse_special_build_check_(actions)/report.html)

- https://github.com/ClickHouse/ClickHouse/pull/34064#issuecomment-1026856098

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -6824,7 +6824,7 @@ bool StorageReplicatedMergeTree::waitForShrinkingQueueSize(size_t queue_size, UI
 bool StorageReplicatedMergeTree::dropPartImpl(
     zkutil::ZooKeeperPtr & zookeeper, String part_name, LogEntry & entry, bool detach, bool throw_if_noop)
 {
-    LOG_TRACE(log, "Will try to insert a log entry to DROP_RANGE for part: " + part_name);
+    LOG_TRACE(log, "Will try to insert a log entry to DROP_RANGE for part {}", part_name);
```
Comment (serxa) [inline src/Storages/StorageReplicatedMergeTree.cpp:6827]: colon missing

**Review by serxa — COMMENTED** [review]
It looks sad that we should add `fmt::runtime` to every std::string. Could just another template specialization for `firstArg` (or any other template) work?

**Comment by azat** [comment]
>It looks sad that we should add fmt::runtime to every std::string.

I thought about this too, but decided that this is not bad, but good, since `fmt::runtime` will suppress format check, and this should be done explicitly (I've found few incorrect formatting already in this PR)

**Comment by serxa** [comment]
Okay, pretty convincing

**Review by serxa — APPROVED** [review]

**Pull request merged**